### PR TITLE
add node mode for flags management

### PR DESCRIPTION
**Pull request by ilovers**
- Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/okex/okexchain/blob/develop/CONTRIBUTING.md#pr-targeting))

- [ ] Linked to github-issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote tests
- [ ] Updated relevant documentation (`docs/`)
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Reviewed `Files changed` in the github PR explorer

______

For Admin Use:
- Added appropriate labels to PR (ex. wip, ready-for-review, docs)
- Reviewers Assigned
- Squashed all commits, uses message "Merge PR #XYZ: [title]" ([coding standards](https://github.com/okex/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: libs/tendermint/mempool/clist_mempool.go [hunk 1]
```diff
@@ -32,6 +32,12 @@ type TxInfoParser interface {
 	GetRawTxInfo(tx types.Tx) ExTxInfo
 }
 
+const (
```
Comment (ylsGit) [inline libs/tendermint/mempool/clist_mempool.go]: Unused constants

File: app/node_mode.go [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+func SetNodeConfig(context *server.Context) {
+	nodeMode := viper.GetString(types.FlagNodeMode)
+	switch types.NodeMode(nodeMode) {
+	case types.RpcNode:
+		setRpcConfig()
+	case types.ValidatorNode:
+		setValidatorConfig()
+	case types.ArchiveNode:
+		setArchiveConfig()
+	case "":
```
Comment (ylsGit) [inline app/node_mode.go:28]: types.NodeMode("")??

File: libs/cosmos-sdk/server/start.go [hunk 3]
```diff
@@ -4,10 +4,11 @@ package server
 
 import (
 	"fmt"
-	"github.com/okex/exchain/libs/tendermint/libs/cli"
 	"os"
 	"runtime/pprof"
 
+	"github.com/okex/exchain/libs/tendermint/libs/cli"
```
Comment (ylsGit) [inline libs/cosmos-sdk/server/start.go]: go import

File: app/node_mode.go (hunk 2 above)
Comment (ilovers) [reply app/node_mode.go:28]: not need, it works.

**Review by ylsGit — DISMISSED** [review]

**Review by banishee — APPROVED** [review]

**Pull request merged**